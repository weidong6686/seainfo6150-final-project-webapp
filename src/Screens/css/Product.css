 /* Product Details */
 .details {
    display: flex;
    justify-content: space-between;
    align-items: flex-start;
    flex-wrap: wrap;
    padding: 1rem;
  }
  .details-image {
    flex: 2 1 60rem;
  }
  .details-image img {
    max-width: 60rem;
    width: 100%;
  }
  .details-info {
    flex: 1 1 30rem;
  }
  .details-action {
    flex: 1 1 30rem;
  }
  
  .details-info ul,
  .details-action ul {
    list-style-type: none;
    padding: 0;
  }
  .details-info li,
  .details-action li {
    margin-bottom: 1rem;
  }
  .back-to-result {
    padding: 1rem;
  }
  .details-action {
    border: 0.1rem #808080 solid;
    border-radius: 0.5rem;
    background-color: #f8f8f8;
    padding: 1rem;
  }
  .details-action ul li:last-child {
    display: flex;
    flex-direction: column;
  }
  .button {
    padding: 1rem;
    border: 0.1rem #808080 solid;
    border-radius: 0.5rem;
    cursor: pointer;
  }
  .button:hover {
    border: 0.1rem #404040 solid;
  }
  .button.primary {
    background-color: #f0c040;
    width: 100%;
  }

  .text-center {
    color: blue;
    text-align: center;
  }

/* Reviews */
ul.review {
  list-style-type: none;
  padding: 0;
}
.review li {
  margin: 1rem;
  margin-bottom: 2rem;
}
.review li > div {
  margin: 0.5rem;
}

/* Rating */
.rating span {
  color: #ffb901;
  font-size: 1.8rem;
  margin: 0.1rem;
}
.rating span:last-child {
  color: #444444;
  font-size: 1.4rem;
}
a > .rating > span:last-child {
  color: #0000ff;
}