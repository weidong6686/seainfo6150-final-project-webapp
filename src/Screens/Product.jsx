import React from 'react';
import data from '../data';
import {Link} from 'react-router-dom';
import './css/Product.css';

const Product = (props) => {
    console.log(props.match.params.id);
    const product = data.products.find(x=> x._id === props.match.params.id);
    return (
    <div>
        <div className = "back-to-result">
            <Link to = "/">Back to result</Link>
        </div>
        <div className = "details">
            <div className = "details-image">
                <img src = {product.image} alt = "product"></img>
            </div>
            <div className = "details-info">
                <ul>
                    <li>
                        <h4>{product.name}</h4>
                    </li>
                    <a href="#reviews">
                    <li>
                        {product.rating} Stars <span>{product.numReviews+' Reviews'}</span>
                    </li>
                    </a>
                    <li>
                        Price: <b>${product.price}</b>
                    </li>
                    <li>
                        Description: FREE Shipping
                        {product.descrption}
                    </li>
                </ul>
            </div>
            <div className = "details-action">
                <ul>
                    <li>
                        Price: ${product.price}
                    </li>
                    <li>
                        Status: {product.status}
                    </li>
                    <li>
                        Qty: <select>
                                <option>1</option>
                                <option>2</option>
                                <option>3</option>
                                <option>4</option>
                            </select>
                    </li>
                    <li>
                        <a href="/cart">
                            <button className="button primary">Add to Cart</button>
                        </a>
                    </li>
                </ul>
            </div>
        </div>
    </div>
    )
}

export default Product;