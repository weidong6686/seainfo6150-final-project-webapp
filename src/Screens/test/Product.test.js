import React from "react";
import Product from "../Product.jsx";
import { BrowserRouter } from 'react-router-dom';
import { render } from "@testing-library/react";
// import data from '../data';

describe("Product tests", () => {
  it("renders correctly", () => {
    const { container } = (<Product />);
    expect(container).toMatchSnapshot();
  });
});