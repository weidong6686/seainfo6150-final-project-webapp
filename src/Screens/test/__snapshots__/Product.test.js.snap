// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product tests renders correctly 1`] = `undefined`;
